main.tf

```diff
@@ -1,13 +1,58 @@
-resource "random_integer" "example" {
+resource "kubernetes_service_account" "service_account" {
   count = module.this.enabled ? 1 : 0
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+  metadata {
+    name        = var.deployment_name
+    namespace   = var.kubernetes_namespace
+    annotations = var.service_account_annotations
   }
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
-}
+resource "kubernetes_deployment" "tfc_cloud_agent" {
+  count = module.this.enabled ? 1 : 0
+
+  metadata {
+    name      = var.deployment_name
+    namespace = var.kubernetes_namespace
+    labels    = var.labels
+  }
+  spec {
+    selector {
+      match_labels = var.selector_match_labels
+    }
+    replicas = var.replicas
+
+    template {
+      metadata {
+        labels      = var.labels
+        annotations = var.deployment_annotations
+      }
+      spec {
+        service_account_name            = kubernetes_service_account.service_account.0.metadata.0.name
+        automount_service_account_token = true
+        container {
+          image = var.agent_image
+          name  = "tfc-agent"
+          env {
+            name  = "TFC_AGENT_TOKEN"
+            value = var.token
+          }
+          env {
+            name  = "TFC_AGENT_NAME"
+            value = module.this.id
+          }
```

Comment:
I see a clear use-case for passing other ENVs for secrets we don't want to expose in TFC (much like the service accounts for AWS IAM). Can we add some optional envs?

Reply:
Also add 

```
         env {
            name  = "TFC_AGENT_LOG_LEVEL"
            value = "..."
          } 
```

Reply:
Here's the full list of envs we should support by default:

```
  -name <name>
    An optional user-specified name for the agent. This name may be used in
    the Terraform Cloud user interface to help easily identify the agent.

    Default: The agent's ephemeral ID, assigned during boot.
    Environment variable: TFC_AGENT_NAME

  -log-level <level>
    The log verbosity expressed as a level string. Level options include
    "trace", "debug", "info", "warn", and "error".

    Default: info
    Environment variable: TFC_AGENT_LOG_LEVEL

  -data-dir <path>
    The path to a directory to store all agent-related data, including
    Terraform configurations, cached Terraform release archives, etc. It is
    important to ensure that the given directory is backed by plentiful
    storage.

    Default: ~/.tfc-agent
    Environment variable: TFC_AGENT_DATA_DIR

  -single
    Enable single mode. This causes the agent to handle at most one job and
    immediately exit thereafter. Useful for running agents as ephemeral
    containers, VMs, or other isolated contexts with a higher-level scheduler
    or process supervisor.

    Default: false
    Environment variable: TFC_AGENT_SINGLE

  -disable-update
    Disable automatic core updates.

    Default: false
    Environment variable: TFC_AGENT_DISABLE_UPDATE

  -address <addr>
    The HTTP or HTTPS address of the Terraform Cloud API.

    Default: https://app.terraform.io
    Environment variable: TFC_ADDRESS

  -token <token>
    The agent token to use when making requests to the Terraform Cloud API.
    This token must be obtained from the API or UI.  It is recommended to use
    the environment variable whenever possible for configuring this setting due
    to the sensitive nature of API tokens.

    Required, no default.
    Environment variable: TFC_AGENT_TOKEN

  -h
    Display this message and exit.

  -v
    Display the version and exit.
```

Reply:
Add the descriptions to each variable description.